### ISSUE TITLE
Enum::key(); casts to a string breaking === comparisons to int keys

### ISSUE THREAD

**Issue by lachlankrautz** [rexlabsio/enum-php#3]
Enum assumes keys are stored as strings and breaks comparisons for int keys.

```
class Schema extends Enum
{
    public const VERSION_01 = 1;
    public const VERSION_02 = 2;
}

$version1 = Schema::VERSION_01();

$version1->is(Schema::VERSION_01);

// throws "Enum or string expected but integer given."
```

**Comment by jodiedunlop** [comment]
I want to make a few BC changes to address this and a couple of other things that are bothering me. I'd like to achieve this by tagging a 2.0.0 release

1. Remove `key()` return signature (so it can have mixed values), and update the `is()` method to do an `is_scalar()` check rather than a string one (fixes above issue reported by lach)
2. Rename `identifier()` to `name()` since this is the _name_ of the constant and identifier is just a dumb name
3. Remove `flip()`, and change the `fromValue()` to use `array_search()` (probably) under the hood rather
4. Probably change `fromValue()` to be `keyForValue()` ... sorry this was my fault, but I guess fromValue is ambiguous as to whether you would get a key, a constant name, or an instance of the enum.

@lachlankrautz @potsed - would appreciate your feedback and/or arguments against the changes plus any other BC changes to roll in now

**Comment by lachlankrautz** [comment]
1. agree 
2. agree
3. agree and for the case of duplicates I think we just shouldn't allow them
4. I think more explicit names are great because we use the enum in a few different contexts. Would be keen for something that all of these ideas very clear:

Getting instance of enum:
- get enum from key
- get enum from name

Getting the value:
- get value from key
- get value from name

Getting from value:
- get key from value
- get name from value

**Comment by jodiedunlop** [comment]
Fixed in #4
